testing/hierarchies.py

```diff
@@ -990,10 +990,14 @@ def _compare_children(hierarchy_type, test, test_child, exhale_child):
                     exhale_grand_child = candidate
                     break
         else:
+            identifier = test_grand_child.breathe_identifier()
             for candidate in exhale_child.children:
-                if candidate.kind == test_grand_child.kind and candidate.name == test_grand_child.name:
+                if candidate.kind == test_grand_child.kind and \
+                        candidate.breathe_identifier() == identifier:
```

Comment:
Seems that with florian's example the child matching search does not work.  Perhaps `ImageBuffer` only works by a fluke.  It may be necessary to start comparing template parameters, but I don't recall if the rest of the testing framework uses them or if they were just stubbed in and never really used.